BEST_PRACTICES.md

```diff
@@ -0,0 +1,146 @@
+
+#### utils
+  - Helpers/Utilities which are not directly related to any host object (that does not need operations on hosts) should be added to utils package.
+  - We have special utility modules for a specific subject. One can add utility helpers in those modules if the helper relates to any of the utlity modules or create new module for new specific subject.
+  - If the helper does not fit into any of the existing utils modules or subject of helper(s) is not big enough to create new , then add in `robottelo/utils/__init__.py` as a standalone helper.
+
+#### hosts.py
+The `hosts.py` module primarily contains three classes that represent RedHat Satellite's three main hosts:
+
+- **Satellite**:
+  - This class serves as the representation of the Satellite host and includes essential properties and methods that describe the Satellite host.
+  - Properties like `hostname` and `port` specify specific details about the Satellite host.
+  - Properties like `api`, `cli`, and `ui_session` provide access to interface objects that expose satellite components through various endpoints. For example, you can access these components as `sat.api.ActivationKey`, `sat.cli.ActivationKey`, or `sat.ui_session.ActivationKey`.
+  - Methods within this class handle satellite operations, especially for reconfiguring the satellite via installation processes etc. Other operational methods are part of the Satellite mixins, including factories. e.g `sat.capsule_certs_generate()`
+  - This class inherits methods from ContentHost and Capsule classes, hence methods from these classes are available with this class objects.
+- **Capsule**:
+  - This class represents the Capsule host and includes essential properties and methods that describe the Capsule host.
+  - Properties like `url` and `is_stream` specify details about the specific Capsule host.
+  - The `satellite` property returns the satellite object to which the Capsule is connected.
+  - Methods within this class manage operations related to the Capsule, especially for reconfiguring it via installation processes. Other operational methods are part of the Capsule mixins. e.g `sat.restart_services()`
+  - This class inherits methods from ContentHost class, hence methods from these classes are available with this class objects.
+- **ContentHost**:
+  - This class represents the ContentHost or Client, including crucial properties and methods that describe these hosts.
+  - Properties like `hostname` specify details about specific ContentHosts.
+  - The `nailgun_host` property returns the API/nailgun object for the ContentHost.
+  - Methods within this class manage operations on ContentHosts, particularly for reconfiguring them. Other operational methods are part of the ContentHost mixins.
+  - This class inherits methods from `Brokers` modules `Host` class, hence methods from that class are available with this class objects.
```

Comment:
I think it would be good to reverse the order of these and state that each higher level inherits the behavior of the previous host class.